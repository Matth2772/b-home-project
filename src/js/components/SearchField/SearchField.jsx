import React from 'react';
import Form from 'react-bootstrap/Form';
import PropTypes from 'prop-types';

function SearchField(props) {
  const { search } = props;
  return (

    <Form.Group controlId="search">
      <Form.Label>Hacker News Search</Form.Label>
      <Form.Control
        type="email"
        placeholder="Search news"
        onChange={e => search(e.target.value)}
      />
    </Form.Group>
  );
}

SearchField.propTypes = {
  search: PropTypes.func,
};

SearchField.defaultProps = {
  search() {
  },
};


export default SearchField;
