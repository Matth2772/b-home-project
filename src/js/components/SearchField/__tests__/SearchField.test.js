/* eslint-disable */
import React from 'react';
import SearchField from '../SearchField';
import renderer from 'react-test-renderer';

test('SearchField rendering well', () => {
  const component = renderer.create(
    <SearchField/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
