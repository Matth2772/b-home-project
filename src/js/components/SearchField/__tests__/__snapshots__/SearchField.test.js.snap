// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField rendering well 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="search"
  >
    Hacker News Search
  </label>
  <input
    className="form-control"
    id="search"
    onChange={[Function]}
    placeholder="Search news"
    type="email"
  />
</div>
`;
